mobile/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -46,121 +46,174 @@ CREATE OR REPLACE PROCEDURE {{.output_schema}}.mk_transaction(DML_STATEMENTS VAR
-          SELECT
-              LISTAGG(
-                  CASE
-                    WHEN isc.data_type='TEXT'
-                      THEN CONCAT(isc.column_name, ' VARCHAR(',isc.character_maximum_length, ')')
-                    WHEN isc.data_type='NUMBER'
-                      THEN CONCAT(isc.column_name, ' NUMBER(', isc.numeric_precision, ',',isc.numeric_scale, ')')
-                    ELSE
-                      CONCAT(isc.column_name, ' ', isc.data_type)
-                  END, '` + delimiter + `')
-              WITHIN GROUP (order by isc.ordinal_position)
-          FROM information_schema.columns AS isc
-          WHERE table_schema = UPPER('` + sch + `')
-            AND table_name = UPPER('` + tbl + `');`;
-
-      var res = snowflake.createStatement({sqlText: stmt}).execute();
-      res.next();
-      result = res.getColumnValue(1);
-
-      return result;
-  }
+    FROM information_schema.columns AS isc
+    WHERE table_schema = UPPER('` + TARGET_SCHEMA + `')
+    AND table_name = UPPER('` + TARGET_TABLE + `')
+    )
 
-  function notIncludedIn(arr) {
-      return function(elt) {
-        return ! arr.includes(elt);
-      };
-  }
+  , source_columns AS (
+    SELECT
+      isc.column_name,
+      isc.data_type,
+      isc.ordinal_position
 
-  function add_columns_to(sch, tbl, cols) {
-      var alter_stmt = `ALTER TABLE ` + sch + `.` + tbl + ` ADD COLUMN ` + cols;
-      snowflake.createStatement({sqlText: alter_stmt}).execute();
+    FROM information_schema.columns AS isc
+    WHERE table_schema = UPPER('` + SOURCE_SCHEMA + `')
+    AND table_name = UPPER('` + SOURCE_TABLE + `')
+    )
 
-      return "ok. Columns added.";
-  }
+  SELECT 
+    SUM(CASE WHEN sc.column_name IS NULL THEN 1 ELSE 0 END) AS missing_in_source,
+    SUM(CASE WHEN tc.column_name IS NULL THEN 1 ELSE 0 END) AS missing_in_target
+
+  FROM target_columns tc
+  FULL OUTER JOIN source_columns sc
+  ON tc.column_name = sc.column_name
+  AND tc.data_type = sc.data_type
+  AND tc.ordinal_position = sc.ordinal_position`
+
+  var column_results = [];
+
+  var res = snowflake.createStatement({sqlText: column_check_stmt}).execute();
+  res.next();
+  missing_in_source = res.getColumnValue(1);
+  missing_in_target = res.getColumnValue(2);
+
+  column_results = [missing_in_source, missing_in_target];
+
+  return column_results;
+
+  $$
+;
+
+
+/*
+  ALTER_TABLE
+  Adds columns from the source table that are missing in the target table.
+  Will only alter the target table if the missing columns occur at the end of the source table.
+  Inputs:
+  SOURCE_SCHEMA:         the schema of the source table
+  SOURCE_TABLE:          the source table to select data from
+  TARGET_SCHEMA:         the schema of the target table
+  TARGET_TABLE:          the target table to delete/insert into
+  To drop:
+  DROP PROCEDURE {{.output_schema}}.alter_table(VARCHAR,VARCHAR,VARCHAR,VARCHAR);
+*/
+CREATE OR REPLACE PROCEDURE {{.output_schema}}.alter_table(SOURCE_SCHEMA VARCHAR,
```

Comment:
I like the idea of separating this out into its own atomic unit. Good design choice.